### PR TITLE
LP-4289 Node12

### PR DESCRIPTION
**Pull request by jjhoughton**
You can ignore everything in deps/include

### CODE REVIEW

File: binding.gyp [hunk 1]
```diff
@@ -24,12 +21,19 @@
             "conditions": [
                 [ "SASL==\"n\"", { "sources!": 
                   ["LDAPSASL.cc", "SASLDefaults.cc"] } ], 
-                [ "SASL==\"y\"", { "sources!": ["LDAPXSASL.cc"] } ]
+                [ "SASL==\"y\"", { "sources!": ["LDAPXSASL.cc"] } ],
+		['OS=="linux" and NODE_VERSION > 9', {
+		    "libraries": [ "../deps/libldap.a", "../deps/liblber.a", "-lresolv", "-lsasl2" ],
```
Comment (jfmesse) [inline binding.gyp]: use only for RH7 or CentOS 7

File: binding.gyp [hunk 2]
```diff
@@ -24,12 +21,19 @@
             "conditions": [
                 [ "SASL==\"n\"", { "sources!": 
                   ["LDAPSASL.cc", "SASLDefaults.cc"] } ], 
-                [ "SASL==\"y\"", { "sources!": ["LDAPXSASL.cc"] } ]
+                [ "SASL==\"y\"", { "sources!": ["LDAPXSASL.cc"] } ],
+		['OS=="linux" and NODE_VERSION > 9', {
```
Comment (jfmesse) [inline binding.gyp]: Does it work for Node 10?

File: binding.gyp [hunk 3]
```diff
@@ -24,12 +21,19 @@
...
+		}, {
+		    "libraries": [ "-lldap" ]
+		}]
             ]
         }
     ],
     "variables": {
-      "SASL": "<!(test -f /usr/include/sasl/sasl.h && echo y || echo n)"
+      "SASL": "<!(test -f /usr/include/sasl/sasl.h && echo y || echo n)",
+      "NODE_VERSION": "<!(node --version | cut -d. -f1 | cut -dv -f2)"
```
Comment (jfmesse) [inline binding.gyp]: extrace RHEL version from /etc/redhat-release if present

File: binding.gyp (hunk 1 above)
Comment (jfmesse) [reply binding.gyp]: and how is libldap.a and the other .a file built? If its something you had to manually do, we should have the method of building it documented or automated in this repo

Comment (jfmesse) [reply binding.gyp]: we need to support RH6, so should definitely contain a binary for that

**Review by jfmesse — COMMENTED** [review]
comments in code

File: binding.gyp (hunk 1 above)
Comment (jjhoughton) [reply binding.gyp]: I have  it automated on napi-ldap you just set the BUILD_OPENLDAP environment variable to 1 and then run yarn and it'll build openssl and openldap for you, see here https://github.com/jjhoughton/napi-ldap/blob/master/binding.gyp. I could either add the same functionality to this project or just link to it from here. That's basically what i did just copy the .a files from the napi-ldap project.

Comment (jjhoughton) [reply binding.gyp]: I'm a little reluctant to put a great deal of effort into this project as i've already fixed so many of the issues with it in napi-ldap. Especially as jeremy childs isn't to responsive when it comes to accepting patches.

File: binding.gyp (hunk 2 above)
Comment (jjhoughton) [reply binding.gyp]: It certainly should although i haven't tested it on node10

**Review by jfmesse — APPROVED** [review]

File: binding.gyp [hunk 4]
```diff
@@ -24,12 +21,33 @@
...
+                        }, {
+                          "libraries": [ "../deps/RHEL7/libldap.a", "../deps/RHEL7/liblber.a" ]
+                        }
+                      ]]
+                    }, {
+                      "libraries": [ "../deps/OSX/libldap.a", "../deps/OSX/liblber.a" ],
+                    }
+                  ]],
+                  "include_dirs": [ "deps/include" ],
+                  "libraries": [ "-lresolv", "-lsasl2" ]
```
Comment (jjhoughton) [inline binding.gyp:40]: Might be a bug here, it looks like this forces you to have the sasl library installed I'll come back to this later

**Pull request merged**